src/VS.Web.CG.Mvc/Identity/IdentityGenerator.cs

```diff
@@ -39,35 +57,89 @@ public IEnumerable<string> TemplateFolders
+                         Path.Combine(DefaultContentRelativeBaseDir, "Data"),
+                         Path.Combine(DefaultContentRelativeBaseDir, "Extensions"),
+                         Path.Combine(DefaultContentRelativeBaseDir, "Services"),
+                         Path.Combine(DefaultContentRelativeBaseDir, "Pages"),
+                         DefaultContentRelativeBaseDir
                      },
                     _projectContext);
             }
         }
 
-        private string _templateFolderRoot;
-        private string TemplateFolderRoot
+        // Returns the set of template folders appropriate for templateModel.ContentVersion
+        private IEnumerable<string> GetTemplateFoldersForContentVersion(IdentityGeneratorTemplateModel templateModel)
         {
-            get
+            if (!(templateModel is IdentityGeneratorTemplateModel2 templateModel2))
+            {   // for back-compat
+                return TemplateFolders;
+            }
+
+            // The default content is packaged under the default location "Identity\*" (no subfolder).
+            if (string.Equals(templateModel2.ContentVersion, ContentVersionDefault, StringComparison.Ordinal))
+            {
+                return TemplateFolders;
+            }
+
+            // For non-default bootstrap versions, the content is packaged under "Identity_Versioned\[Version_Identifier]\*"
+            // Note: In the future, if content gets pivoted on things other than bootstrap, this logic will need enhancement.
+            if (string.Equals(templateModel2.ContentVersion, ContentVersionBootstrap3, StringComparison.Ordinal))
             {
-                if (string.IsNullOrEmpty(_templateFolderRoot))
+                return TemplateFoldersUtilities.GetTemplateFolders(
+                    Constants.ThisAssemblyName,
+                    _applicationInfo.ApplicationBasePath,
+                    new[] {
+                    Path.Combine(VersionedContentRelativeBaseDir, $"Bootstrap{templateModel2.BootstrapVersion}")
+                    },
+                    _projectContext);
+            }
+
+            // This should get caught by IdentityGeneratorTemplateModelBuilder.ValidateCommandLine() and emit the same error. 
+            // But best to be safe here.
+            // Note: If we start pivoting content on things other than bootstrap version, this error message will need to be reworked.
+            throw new InvalidOperationException(string.Format(MessageStrings.InvalidBootstrapVersionForScaffoldingIdentity, templateModel2.BootstrapVersion, string.Join(", ", ValidBootstrapVersions)));
+        }
+
+        // Returns the root directory of the template folders appropriate for templateModel.ContentVersion
+        private string GetTemplateFolderRootForContentVersion(IdentityGeneratorTemplateModel templateModel)
+        {
+            string relativePath = null;
+
+            if (templateModel is IdentityGeneratorTemplateModel2 templateModel2)
```

Comment:
Agree that it's a less than ideal name. I needed something that extends IdentityGeneratorTemplateModel - that class can't be modified because it's a public class and this is a minor release, and the "2" is a simple naming convention. I could have gone with something like IdentityGeneratorTemplateModelExtended, but then if we need to extend it again before another major release, the naming gets pretty awkward.

For the 3.0 release, the properties in IdentityGeneratorTemplateModel2 (and possibly 3, 4, etc.) will get absorbed into IdentityGeneratorTemplateModel (because we can make breaking changes then).

Regarding the syntax, yes this works. It's relatively new, I don't recall exactly when it became available.

Reply:
Good point. I can't actually come up with a better name now that you point out the restrictions and possibilities.